express-auth/routes/index.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+
+// User model
+const User = require('../models/user');
+
+// BCrypt to encrypt passwords
+const bcrypt = require('bcrypt');
+const bcryptSalt = 10;
+
+router.get('/welcome', (req, res, next) => {
+  res.render('welcome');
+});
+
+router.get('/login', (req, res, next) => {
+  res.render('auth/login');
+});
+
+router.post('/login', (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  if (username === '' || password === '') {
+    res.render('auth/login', {
+      errorMessage: 'Indicate a username and a password to sign up'
+    });
+    return;
+  }
+
+  User.findOne({ 'username': username })
+    .then(result => {
+      if (!result) {
+        res.render('auth/login', {
+          errorMessage: "The username doesn't exist"
+        });
+      } else if (bcrypt.compareSync(password, result.password)) {
+        // here says user undefined and crash!!!!!HELP!!!!!
+        req.session.user = result;
```

Comment:
I was not able to reproduce this. Did you fix it in the meantime?

Reply:
Yes fixed!